From shemminger@osdl.org  Mon Nov 13 15:17:05 2006
Date: Mon, 13 Nov 2006 15:12:45 -0800
From: Stephen Hemminger <shemminger@osdl.org>
To: Greg KH <greg@kroah.com>
Cc: linux-usb-devel@lists.sourceforge.net
Subject: USB: resume_device symbol conflict
Message-ID: <20061113151245.168ea0f3@localhost>
Mime-Version: 1.0
Content-Type: text/plain; charset=US-ASCII

There are two definitions of resume_device. A global one is defined
in resume.c, and a local one is defined in USB core. The linker appears
to do the right thing, but it is bad practice and makes backtrace
decoding harder.

Signed-off-by: Stephen Hemminger <shemminger@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/driver.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/driver.c
+++ gregkh-2.6/drivers/usb/core/driver.c
@@ -837,7 +837,7 @@ done:
 }
 
 /* Caller has locked udev's pm_mutex */
-static int resume_device(struct usb_device *udev)
+static int usb_resume_device(struct usb_device *udev)
 {
 	struct usb_device_driver	*udriver;
 	int				status = 0;
@@ -1109,14 +1109,14 @@ int usb_resume_both(struct usb_device *u
 				status = -EHOSTUNREACH;
 		}
 		if (status == 0)
-			status = resume_device(udev);
+			status = usb_resume_device(udev);
 		if (parent)
 			usb_pm_unlock(parent);
 	} else {
 
 		/* Needed only for setting udev->dev.power.power_state.event
 		 * and for possible debugging message. */
-		status = resume_device(udev);
+		status = usb_resume_device(udev);
 	}
 
 	/* Now the parent won't suspend until we are finished */
